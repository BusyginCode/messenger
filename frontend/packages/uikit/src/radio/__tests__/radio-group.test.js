import React from 'react'
import { shallow } from 'enzyme'
import RadioGroup from '../radio-group'

describe('uikit/RadioGroup', () => {
  it('renders correctly', () => {
    const options = [{ value: 'val1', label: 'Value1' }, { value: 'val2', label: 'Value2' }]
    const onChange = jest.fn()

    const wrapper = shallow(<RadioGroup options={options} value="val1" onChange={onChange} />)

    expect(wrapper).toMatchSnapshot()

    wrapper
      .find('Radio')
      .at(0)
      .prop('onChange')({ target: { value: 'val1' } })

    expect(onChange).toHaveBeenCalledTimes(1)
  })
})
