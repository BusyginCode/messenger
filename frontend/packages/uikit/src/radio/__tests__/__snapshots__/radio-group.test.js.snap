// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uikit/RadioGroup renders correctly 1`] = `
<div
  className="radioGroup inline"
>
  <label
    className="radioWrapper"
    key="val1"
  >
    <Radio
      checked={true}
      data-role="value_val_1"
      onChange={[Function]}
      value="val1"
    />
    <AText
      breed="body"
      className="label"
      data-role="value_1_val1_0"
      semibold={true}
      size="l"
    >
      Value1
    </AText>
  </label>
  <label
    className="radioWrapper"
    key="val2"
  >
    <Radio
      checked={false}
      data-role="value_val_2"
      onChange={[Function]}
      value="val2"
    />
    <AText
      breed="body"
      className="label"
      data-role="value_2_val2_1"
      semibold={false}
      size="l"
    >
      Value2
    </AText>
  </label>
</div>
`;
